lib/engine/step/token_merger.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module Engine
+  module Step
+    module TokenMerger
+      def tokens_in_same_hex(surviving, others)
+        # Are there tokens in the same hex?
+        # Most will be released by remove_duplicate_tokens, but NY style tiles need to be solved by the user
+        (surviving.tokens.map { |t| t.city&.hex } & others_tokens(others).map { |t| t.city&.hex }).any?
+      end
+
+      def tokens_above_limits?(surviving, others)
+        tokens_in_same_hex(surviving, others) ||
+        (surviving.tokens + others_tokens(others)).select(&:used).size > @game.class::LIMIT_TOKENS_AFTER_MERGER
+      end
+
+      def others_tokens(others)
+        others = Array(others)
+        others.flat_map(&:tokens)
+      end
+
+      def remove_duplicate_tokens(surviving, others)
+        # If there are 2 station markers on the same city the
+        # surviving company must remove one and place it on its charter.
+        # In the case of NY tiles this is ambigious and must be solved by the user
+
+        others = others_tokens(others).map(&:city).compact
+        surviving.tokens.each do |token|
```

Comment:
do you need to do a dup here? dose remove! mutate tokens?

Reply:
huh?

Reply:
what dup? Sorry unsure what you mean

Reply:
surviving.tokens.dup.each do 

because you do a remove right?

Reply:
oh, i see what you mean. no the token.remove! just un-places the token. (This is exactly the same as the original 1817 code)